### DISCUSSION TITLE
ScriptReference/AssetDatabase.FindAssets

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#30] [Notes]
# ScriptReference/AssetDatabase.FindAssets

https://docs.unity3d.com/ScriptReference/AssetDatabase.FindAssets.html

**Comment by vertxxyz** [comment]
Use `"Assets"` and `"Packages"` in the `searchInFolders` parameter to search in both.

**Comment by vertxxyz** [comment]
When using the type search `t:ObjectType` some types will not be found. If this is the case you should provide the [FullName](https://docs.microsoft.com/en-us/dotnet/api/system.type.fullname).
This will work for types that were previously already being found, so it may be advisable to always use instead of the type [Name](https://docs.microsoft.com/en-us/dotnet/api/system.type.name).

**Comment by docscomments** [comment]
**Comment by `@fdegryse`:**  

`FindAssets` will also list sub assets but because sub assets do not have their own GUID, the GUID of their respective main asset will be listed instead.  
If multiple sub assets from the same main asset are listed, its GUID will be listed multiple times.

**Comment by docscomments** [comment]
**Comment by `@Kerfuffles`:**  

Besides `t:` and `l:`, other possible values are:  
`v:` Can be used to find objects with different states. e.g. `v:modified` will show objects that are locally modified  
`s:` Searches by soft lock state. e.g. `s:inprogress` will show objects modified by anyone, except you  
`a:` Acts as a filter. Possible values are `all` (*all folders*), `assets` (only *Assets/*), and `packages` (only *Packages/*)  
`b:` Searches for assets in asset bundles, by the asset bundle name. e.g. `b:mybundle` will only look in `mybundle` asset bundle  
`ref:` Searches by instance ID  
`glob:` Can be used to find assets with extensions of any type, e.g. `glob:Assets/**/*.{png|PNG}`, which will show any object ending with .png or .PNG in any subfolder.

**Comment by FreyaHolmer** [comment]
If the second parameter `searchInFolders` is an empty array, it will return all assets in the observable universe, instead of no assets, so be very careful with that one lol
